pkg/mesh/topology.go

```diff
@@ -379,13 +408,13 @@ func findLeader(nodes []*Node) int {
 	var leaders, public []int
 	for i := range nodes {
 		if nodes[i].Leader {
-			if isPublic(nodes[i].Endpoint.IP) {
+			if isPublic(nodes[i].Endpoint.IP()) {
```

Comment:
```suggestion
			if !net.IsPrivate(nodes[i].Endpoint.IP()) {
```

Reply:
Sgtm, I think this is relatively new to the stdlib so nice to switch to it

Reply:
we would have to bump go version first :( 
Didn't know it was new.

Reply:
Cool cool, we can always do it in a follow-up